profiles/views.py

```diff
@@ -61,28 +59,35 @@ class ProfileList(ListCreateAPIView):
      include_all: bool.
     """
     serializer_class = ProfileSerializer
-    permission_classes = (IsAuthenticatedOrReadOnly, )
+    permission_classes = [IsAuthenticatedOrReadOnly & SavedCompaniesListPermission]
     pagination_class = ForumPagination
 
     def get_queryset(self):
+        query_params_serializer = QueryParamSerializer(data=self.request.query_params)
         company_type = self.request.query_params.get("company_type")
         activity_type = self.request.query_params.get("activity_type")
+        userid = self.request.query_params.get("userid")
         HEADER_ACTIVITIES = ["producer", "importer", "retail", "horeca"]
 
-        if self.request.query_params:
+        queryset = Profile.objects.filter(is_deleted=False).order_by("profile_id")
+       
+        if userid:
             try:
-                userid = self.request.query_params.get('userid')
-                return Profile.objects.filter(person_id=userid)
+                return queryset.filter(person_id=userid)
             except ValueError:
-                raise ValidationError(detail='Bad request')
+                pass
         if company_type == "startup":
-            return Profile.objects.filter(comp_is_startup=True).order_by("profile_id")
+            return queryset.filter(comp_is_startup=True)
         elif company_type == "company":
-            return Profile.objects.filter(comp_registered=True).order_by("profile_id")
+            return queryset.filter(comp_registered=True)
         if activity_type in HEADER_ACTIVITIES:
-            return Profile.objects.filter(comp_activity__name=activity_type).order_by("profile_id")
-
-        return Profile.objects.filter(is_deleted=False).order_by("profile_id")
+            return queryset.filter(comp_activity__name=activity_type)
```

Comment:
Too many returns in one function, especially if you will want to use multiple filters

queryset = queryset.filter(comp_activity__name=activity_type)